lib/charms/traefik_k8s/v1/ingress_per_unit.py

```diff
@@ -586,159 +609,146 @@ def __init__(
         All request args must be given as keyword args.
 
         Args:
-            charm: the charm that is instantiating the library.
-            relation_name: the name of the relation name to bind to
+            `charm`: the charm that is instantiating the library.
+            `relation_name`: the name of the relation name to bind to
                 (defaults to "ingress-per-unit"; relation must be of interface
                 type "ingress_per_unit" and have "limit: 1")
-            host: Hostname to be used by the ingress provider to address the
-            requirer unit; if unspecified, the pod ip of the unit will be used
-            instead
-        Request Args:
-            port: the port of the service
-        """
+            `host`: Hostname to be used by the ingress provider to address the
+                requirer unit; if unspecified, the FQDN of the unit will be
+                used instead
+            `port`: port to be used by the ingress provider to address the
+                    requirer unit.
+            `listen_to`: Choose which events should be fired on this unit:
+                "only-this-unit": this unit will only be notified when ingress
+                  is ready/revoked for this unit.
+                "all-units": this unit will be notified when ingress is
+                  ready/revoked for any unit of this application, including
+                  itself.
+                "all": this unit will receive both event types (which means it
+                  will be notified *twice* of changes to this unit's ingress!)
```

Comment:
when would this mode be useful?

Reply:
Maybe you want a specialized handler for when this unit's url changes, and a generic handler for when the url of any peer unit (including yourself) does.

But in general it felt wrong to constrain the API to only one of the two choices. I wanted to leave the option open to 'just fire all events'.